internal/pkg/service/buffer/worker/upload/retry_test.go

```diff
@@ -58,3 +82,283 @@ func TestRetryAt(t *testing.T) {
 	assert.Equal(t, "2010-01-01T08:50:00Z", upload.RetryAt(b, now, 6).Format(time.RFC3339))
 	assert.Equal(t, "2010-01-01T11:50:00Z", upload.RetryAt(b, now, 7).Format(time.RFC3339))
 }
+
+func TestRetryFailedUploadsTask(t *testing.T) {
+	t.Parallel()
+
+	// Test dependencies
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	clk := clock.NewMock()
+	clk.Set(time.Time{}.Add(time.Second))
+	etcdNamespace := "unit-" + t.Name() + "-" + gonanoid.Must(8)
+	client := etcdhelper.ClientForTestWithNamespace(t, etcdNamespace)
+	opts := []dependencies.MockedOption{
+		dependencies.WithClock(clk),
+		dependencies.WithEtcdNamespace(etcdNamespace),
+	}
+
+	// Create file
+	file := &storageapi.File{
+		Name:           "slice-upload-task-test",
+		IsSliced:       true,
+		Provider:       s3.Provider,
+		Region:         "us‑east‑2",
+		S3UploadParams: &s3.UploadParams{Key: "foo", Bucket: "bar"},
+	}
+
+	// Create receivers, exports and records
+	apiDeps1 := bufferDependencies.NewMockedDeps(t, append(opts, dependencies.WithUniqueID("api-node-1"))...)
+	str := apiDeps1.Store()
+	sliceKey := createExport(t, "my-receiver-1", "my-export-1", ctx, clk, client, str, file)
+	createRecords(t, ctx, clk, apiDeps1, sliceKey.ReceiverKey, 1, 5)
+	clk.Add(time.Minute)
+
+	// Requests to the file storage will fail
+	uploadTransport := httpmock.NewMockTransport()
+	uploadTransport.RegisterNoResponder(func(request *http.Request) (*http.Response, error) {
+		return nil, notRetryableError{error: errors.New("some network error")}
+	})
```

Comment:
Upload rozbijeme nejakou chybou, co nema zmysel opakovat (aby to bolo rychlejsie otestovane).